fea-rs/src/compile/lookups.rs

```diff
@@ -430,8 +430,12 @@ impl AllLookups {
     }
 
     /// Returns `true` if there is an active lookup of this kind
-    pub(crate) fn has_current_kind(&self, kind: Kind) -> bool {
-        self.current.as_ref().map(SomeLookup::kind) == Some(kind)
+    pub(crate) fn has_current_kind(&self, kind: Kind, flags: LookupFlagInfo) -> bool {
```

Comment:
instead of changing this method I think I'd prefer to add a separate method with a name like `has_same_flags(&self, flags: LookupFlagInfo) -> bool`. This will let us continue to use this function in places where we don't care about the flags (like in the promotion logic below) and then in places where we can we can just write, `if !self.lookups.has_current_kind(kind) && !self.lookups.has_same_flags(flags) { ..` in the places where we care about both.